modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/ClickGoalPicker/ClickGoalPicker.es.js

```diff
@@ -370,25 +370,20 @@ function OverlayContainer({allowEdit, root}) {
 
 	useEventListener('keydown', handleKeydown, true, document);
 
-	const handleMouseDown = useCallback(() => {
-		mousedownRef.current = true;
-	}, []);
-
-	const handleMouseUp = useCallback(
+	const handleClick = useCallback(
 		(event) => {
-			if (mousedownRef.current === true) {
+			if (clickRef.current === true) {
```

Comment:
If you move this event from `mousedown` to `click`, this check would likely cause some issues, as the first click on the page would have to happen first before this ref is set to true.